### PR TITLE
Add rulesetInterface

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
This is the interface to access the ruleset.

### CODE REVIEW

File: Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * Get all possible divisions from the ruleset.
+     * @param language the preferred language for translation
+     * @return a {@link Map} collection of all divisions that contains {@link String} as id and {@link String} as translation
+     */
+    Map<String,String> getAllDivisions(LanguageRange language);
+
+    /**
+     * Get all possible acquisition stages.
+     * @return A Collection of all acquisition stages
```
[the remark this answers is not in the document]
Comment (Beacze) [reply Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java]: Actually Collection with uppercase is here correct:
`Collection<String> getAllAcquisiionStages();`

File: Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     * Get all metadata, which can be added for a specific division.
+     * @param rulesetRequestData The needed data for retrieval.
+     * @return A {@link Map} collection of addable metadata that contains {@link String} as id and {@link String} as translation
+     */
+    Map<String, String> getAddableMetadataForDivision(RulesetRequestData rulesetRequestData);
+
+    /**
+     * Gets the sorted visible metadata.
+     * @param rulesetRequestData The needed data for retrieval.
+     * @return a Collection of Metadata.
```
[the remark this answers is not in the document]
Comment (Beacze) [reply Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java]: Actually Collection with uppercase is here correct:
`Collection<Metadata> getSortedVisibleMetadataForDivision(RulesetRequestData rulesetRequestData);`

File: Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    Collection<Metadata> getSortedVisibleMetadataForDivision(RulesetRequestData rulesetRequestData);
```
[the remark this answers is not in the document]
Comment (Kathrin-Huber) [reply Kitodo-API/src/main/java/org/kitodo/api/rulesetreader/RulesetReaderInterface.java]: How do you assume, that the returned metadata is those that were previously passed in when the method is called "getSortedVisibleMetadataForDivision"
It returns the sorted visible metadata for the given division.

We discussed the other stuff at the hackathon, when we (together) designed the interface.

**Comment by Kathrin-Huber** [comment]
We discussed this interface at the hackathon. If there is some functionality missing, we will find it out, when using the interface. as i remember, we covered a huge amount of functionality

Stuff like validation etc is made in another module and does not have to be covered by this interface.

**Pull request merged**